### DISCUSSION TITLE
How to create my own lookup editor?

### DISCUSSION THREAD

**Discussion by imhemayatsangin** [serenity-is/Serenity#6002] [Help] [answered]
I want to create a dropdown button or select box where i will make my own querry having some kind of filtering for example:
 I want to display Employees in a dropdown which are only doctors and my SQL query will be something like this.
```
SELECT e.EmpID,e.EmpName FROM Employee AS e
WHERE EmpTypeID=3
```
Here the EmpTypeID=3 is for Doctor. So how i will implement it in the serenity framework? Please i'm new as a learner so if anybody can explain it clearly how to create custom lookup editor for this and guide me little bit.
Looking forward for your help.
Thanks

**Comment by arunkumar12712** [comment]
https://serenity.is/docs/tutorials/multi_tenancy/handling_lookup_scripts

**Comment by luongnv6** [comment, marked as answer]
```C#
[LookupScript("Default.Doctor", Expiration = -1, Permission = "?"), ScriptInclude]
    public class DoctorLookup : LookupScript
    {
        public DoctorLookup()
        {
            this.IdField = "EmpID";
            this.TextField = "EmpName";
            this.getItems = this.GetItems;
        }
        protected IEnumerable<dynamic> GetItems()
        {
            using (var con = SqlConnections.NewFor<EmployeeRow>())  //EmployeeRow use create connection
            {
                return con.Query(@"
SELECT e.EmpID,e.EmpName FROM Employee AS e
WHERE EmpTypeID=3
", null, commandType: CommandType.Text);
            }
        }
    }
```

**Reply by imhemayatsangin** [reply]
Dear luongnv6, where should i put these lines of code? in ts file? and how can i access in the form? please if you can guide me with little bit explanation i will be so much thankful of you!.

**Reply by edwardch** [reply]
Check out something like this: [Serenity.Demo.Northwind/Customer/CustomerCountryLookup.cs](https://github.com/serenity-is/common-features/blob/45957bbdfbaca522c632e0172e88506c0d8cb015/src/Serenity.Demo.Northwind/Serenity.Demo.Northwind/Customer/CustomerCountryLookup.cs)

**Reply by imhemayatsangin** [reply]
Thank you so much bro
